plugins/index.js

```diff
@@ -77,12 +88,32 @@ export function loadPlugin(manifest) {
     script.type = 'text/javascript';
     script.src = getSiteURL() + bundlePath;
     script.onload = onLoad;
+    console.log('Loading ' + manifest.id + ' plugin'); //eslint-disable-line no-console
     document.getElementsByTagName('head')[0].appendChild(script);
 }
 
+// initializePlugin creates a registry specific to the plugin and invokes any initialize function
+// on the registered plugin class.
+function initializePlugin(manifest) {
+    // Initialize the plugin
+    const plugin = window.plugins[manifest.id];
+    const registry = new PluginRegistry(manifest.id);
+    if (plugin.initialize) {
+        plugin.initialize(registry, store);
+    }
+}
+
+// removePlugin triggers any uninitialize callback on the registered plugin, unregisters any
+// event handlers, and removes the plugin script from the DOM entirely. The plugin is responsible
+// for removing any of its registered components.
 export function removePlugin(manifest) {
+    console.log('Removing ' + manifest.id + ' plugin'); //eslint-disable-line no-console
     const plugin = window.plugins[manifest.id];
-    if (plugin && plugin.deinitialize) {
+    if (plugin && plugin.uninitialize) {
+        plugin.uninitialize();
+
+    // Support the deprecated deinitialize callback from the plugins beta.
+    } else if (plugin && plugin.deinitialize) {
```

Comment:
Supporting `deinitialize`, but we've converged on `un` elsewhere so this made sense for consistency.